messaging/broker.go

```diff
@@ -352,32 +351,32 @@ func copyFileN(w io.Writer, path string, n int64) (int64, error) {
 	return io.CopyN(w, f, n)
 }
 
-// Restore reads the broker state.
-func (b *Broker) Restore(r io.Reader) error {
+// ReadFrom reads a broker snapshot from r.
```

Comment:
It's the number of bytes read but we're not supporting it. It's just an implementation of the [`io.ReaderFrom`](http://golang.org/pkg/io/#ReaderFrom) interface. We return an error for any partial reads.